### PR TITLE
Refactor and clean up hf_ptq.py

### PR DESCRIPTION
**Pull request by shengliangxu**
## What does this PR do?

Refactor and clean up hf_ptq.py

This script has several separate logic and the code of them are entangled, making it really hard to add new features

Refactor them so that we separate these logics:

1. sparsity, all logic go to sparsity_main. TODO: we may actually move this logic out to a separate script

2. quantize, all logic go to quantize_main.

   2.1 plain quantization with a single quantization format

   2.2 auto quantization

In the quantization pipeline, separate the pipeline to:

1. model loading
2. calibrate dataset loading
3. pre-quantize processing
4. actual quantize
5. post-quantize processing
6. quantized model export

## Testing

tested the mono quantization:

```
python examples/llm_ptq/hf_ptq.py \
    --pyt_ckpt_path=Qwen/Qwen3-8B \
    --export_path=qwen3-8B_fp8 \
    --qformat=fp8 \
    --kv_cache_qformat=fp8 \
    --calib_size=16 \
    --batch_size=0 \
    --trust_remote_code \
    --export_fmt=hf
```

and deployed to vLLM and TRTLLM, validated accuracy using lm_eval

tested auto quantize:

```
python examples/llm_ptq/hf_ptq.py \
    --qformat=nvfp4,fp8 \
    --auto_quantize_score_size 128 \
    --auto_quantize_bits 5.0 \
    --auto_quantize_checkpoint llama-8B-auto-quantize-checkpoint \
    --pyt_ckpt_path=meta-llama/Meta-Llama-3-8B \
    --export_path=llama-8B_auto_quantize
    --kv_cache_qformat=fp8 \
    --calib_size=16 \
    --batch_size=0 \
    --trust_remote_code \
    --export_fmt=tensorrt_llm
```
and compared exported files

### CODE REVIEW

File: examples/llm_ptq/multinode_ptq.py [hunk 1]
```diff
@@ -332,7 +332,6 @@ def main(args):
         args.qformat,
         args.kv_cache_qformat,
         args.awq_block_size,
-        None,
```
Comment (cjluo-nv) [inline examples/llm_ptq/multinode_ptq.py:335]: @shengliangxu is this intended?

File: examples/llm_ptq/hf_ptq.py [hunk 2]
```diff
@@ -364,258 +342,130 @@ def main(args):
...
+    )
+    full_model = mts.sparsify(
+        full_model,
+        args.sparsity_fmt,
+        config={"data_loader": calib_dataloader, "collect_func": lambda x: x},
+    )
+    mts.export(full_model)
+
+
+def plain_quantize(
```
Comment (cjluo-nv) [inline examples/llm_ptq/hf_ptq.py]: nit: maybe let's call default_quantize or single_precision_quantize?

File: examples/llm_ptq/multinode_ptq.py (hunk 1 above)
Comment (shengliangxu) [reply examples/llm_ptq/multinode_ptq.py:335]: yes, this None is for auto_quantize, the target function does not apply to auto_quantize at all, so removed this useless arg.

File: examples/llm_ptq/hf_ptq.py (hunk 2 above)
Comment (shengliangxu) [reply examples/llm_ptq/hf_ptq.py]: It's not necessary single precision, just a single config. We can have mixed precision even using single config.

default_quantize does not sounds a bit too general and a bit tedious. How about mono_quantize? meaning quantize using a single config

Comment (shengliangxu) [reply examples/llm_ptq/hf_ptq.py]: auto_quantize vs mono_quantize, quite symmetric

**Comment by shengliangxu** [comment]
> Thanks @shengliangxu for the refactoring. Could you also validate that the checkpoint before and after this change is the same?

done, updated the description.

File: examples/llm_ptq/hf_ptq.py [hunk 3]
```diff
@@ -701,11 +550,250 @@ def output_decode(generated_ids, input_shape):
...
+    args: argparse.Namespace,
+    full_model: torch.nn.Module,
+    model_type: str | None,
+    tokenizer: PreTrainedTokenizerBase | None,
+    processor: BaseImageProcessor | ProcessorMixin | None,
+    preview_input_ids,
+    generated_ids_before_ptq,
+    is_nemotron_vl_model,
+    first_text_speech_dataset,
+):
```
Comment (meenchen) [inline examples/llm_ptq/hf_ptq.py:606]: nit: Could you add descriptions for this function?

Comment (shengliangxu) [reply examples/llm_ptq/hf_ptq.py:606]: done.

**Comment by kaix-nv** [comment]
LGTM.
